mm2src/mm2_io/src/fs.rs

```diff
@@ -276,10 +271,51 @@ where
     read_files_with_extension(dir_path, "json").await
 }
 
+/// Creates all the directories along the path to a file.
+pub fn create_parents(path: &dyn AsRef<Path>) -> IoResult<()> {
+    let parent_dir = path.as_ref().parent();
+    let Some(parent_dir) = parent_dir else {
+        return MmError::err(
+            io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("{} has no parent directory", path.as_ref().display()),
+            ))
+    };
+    if parent_dir.exists() {
+        if !parent_dir.is_dir() {
+            return MmError::err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                format!("{} is not a directory", parent_dir.display()),
+            ));
+        }
+    } else {
+        fs::create_dir_all(parent_dir)?;
+    }
+    Ok(())
+}
+
+pub fn write(path: &dyn AsRef<Path>, content: &[u8], use_tmp_file: bool) -> IoResult<()> {
+    // Create all the directories in the path.
+    create_parents(path)?;
+    let path_tmp = if use_tmp_file {
+        PathBuf::from(format!("{}.tmp", path.as_ref().display()))
+    } else {
+        path.as_ref().to_path_buf()
+    };
+    // Write the file content into the temp file and then rename the temp file into the desired name.
+    fs::write(&path_tmp, content)?;
+    if use_tmp_file {
+        fs::rename(&path_tmp, path.as_ref())?;
+    }
```

Comment:
hmmm if `fs::rename fails`, the temporary file will remain. 
I suggest at least logging a warning to indicate that the temporary file was not removed.

Reply:
I hate this kind of magical functions. I'd rather handle renaming explicitly wherever needed. With this `write` function, it's unclear what's happening.

Reply:
> ummm, imo, i think that's very low level to include explicitly in callers of write(). i would rather abstract this inside mm2_io instead.

At least make it clear that it isn't just a `write` to avoid confusion then.

Reply:
@onur-ozkan [bb47ee2](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2398/commits/bb47ee265aae02538e1469c41848da3faa21a8af)
@laruh [b3b3384](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2398/commits/b3b33843cabfe7541e6a35a151cf41542343d30b)

Reply:
Onur please resolve this, if your notes are fixed